### PR TITLE
Add UCN documentation [CN-1131]

### PR DESCRIPTION
**Pull request by SeriyBg**

### CODE REVIEW

File: docs/modules/ROOT/pages/user-code-namespaces.adoc [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+[source,yaml,subs="attributes+"]
+----
+include::ROOT:example$/hazelcast-ucn.yaml[]
+----
+
+== Creating a new User Code Namespace
+
+To create a new User Code Namespace you must create a new `UserCodeNamespace` CR. The name of the CR will be used as the name of the created User Code Namespace. The JAR files stored in the configured bucket will be uploaded to the Hazelcast cluster under the newly created namespace.
+
+.Example of User Code Namespaces
```
Comment (kutluhanmetin) [inline docs/modules/ROOT/pages/user-code-namespaces.adoc:22]: Shall we add multiple tabs for different cloud providers (AWS, GCP, Azure) as we do in backup and restore docs?

File: docs/modules/ROOT/pages/user-code-namespaces.adoc [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+To enable User Code Namespaces you must configure the `userCodeNamespaces` section in the Hazelcast custom resource.
+
+.Example of Hazelcast with User Code Namespaces enabled
+[source,yaml,subs="attributes+"]
+----
+include::ROOT:example$/hazelcast-ucn.yaml[]
+----
+
+== Creating a new User Code Namespace
```
Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/user-code-namespaces.adoc:18]:
```suggestion
== Creating a User Code Namespace
```

File: docs/modules/ROOT/pages/user-code-namespaces.adoc [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+To create a new User Code Namespace you must create a new `UserCodeNamespace` CR. The name of the CR will be used as the name of the created User Code Namespace. The JAR files stored in the configured bucket will be uploaded to the Hazelcast cluster under the newly created namespace.
```
Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/user-code-namespaces.adoc:20]:
```suggestion
To create a User Code Namespace, you must create a new `UserCodeNamespace` CR. The name of the CR is used as the name of the created User Code Namespace. The JAR files stored in the configured bucket are uploaded to the Hazelcast cluster under the newly created namespace.
```

File: docs/modules/ROOT/pages/user-code-namespaces.adoc [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+.Example of User Code Namespaces
+[source,yaml,subs="attributes+"]
+----
+include::ROOT:example$/user-code-namespace.yaml[]
+----
+<1> The bucket URI where backup data will be stored in
```
Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/user-code-namespaces.adoc:27]:
```suggestion
<1> The bucket URI in which to store backup data
```

**Review by rebekah-lawrence — CHANGES_REQUESTED** [review]

**Review by hasancelik — CHANGES_REQUESTED** [review]
@SeriyBg I think we need add executor services and mapstore related parts with UCN page, as we did for UCD before. WDYT?
https://docs.hazelcast.com/operator/latest/user-code-deployment

File: docs/modules/ROOT/pages/user-code-deployment.adoc [hunk 5]
```diff
@@ -1,6 +1,8 @@
 = User Code Deployment
 :description: Hazelcast allows you to deploy your own custom and domain classes to cluster members. From Hazelcast Platform Operator, you can deploy your application code from external buckets and ConfigMap objects in Kubernetes.
 
+WARNING: User Code Delpoymen is deprecated. Use xref:user-code-namespaces.adoc[User Code Namespaces] instead.
```
Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/user-code-deployment.adoc:4]:
```suggestion
WARNING: User Code Delpoyment is deprecated. Use xref:user-code-namespaces.adoc[User Code Namespaces] instead.
```

File: docs/modules/ROOT/pages/user-code-namespaces.adoc [hunk 6]
```diff
@@ -1,34 +1,64 @@
...
++
+--
+User Code Namespaces in Azure Blob:
+[source,yaml,subs="attributes+"]
 ----
+include::ROOT:example$/user-code-namespace-azure.yaml[]
+----
+--
+====
+<1> The bucket URI in which to store backup data
```
Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/user-code-namespaces.adoc:55]:
```suggestion
<1> The bucket URI in which to store backup data.
```

**Review by rebekah-lawrence — COMMENTED** [review]
Small typo only

File: docs/modules/ROOT/pages/user-code-deployment.adoc (hunk 5 above)
Comment (hasancelik) [reply docs/modules/ROOT/pages/user-code-deployment.adoc:4]: I think it should be *Deployment* 🙂

**Review by hasancelik — APPROVED** [review]

File: docs/modules/ROOT/pages/user-code-deployment.adoc (hunk 5 above)
Comment (rebekah-lawrence) [reply docs/modules/ROOT/pages/user-code-deployment.adoc:4]: Good catch Hasan! Hey, it is Friday afternoon - suggestion updated

Comment (SeriyBg) [reply docs/modules/ROOT/pages/user-code-deployment.adoc]: Thanks for the catch! Sorry @rebekah-lawrence, I missed that you've posted a suggestion and pushed the change manually 🙂

**Review by rebekah-lawrence — APPROVED** [review]
No problem, was just a typo ;-)

LGTM

**Pull request merged**